Fixes issue #43 where files are displayed in the browser instead of downloading.

bkolden opened the pull request:
This change forces all file transfers to have the Content-Type set to "application/octet-stream" resulting in the file downloading instead of being displayed in the browser.

npes87184 commented:
Hi, thank you for tihs.

It seen to me that this should be an option and default on.

Usually, I create simple web server on a git-diff file using this tool for my colleague to review. Although this situation is not common, I suppose that display file on browser is a good choice in this case.

bkolden commented:
So, do you think I should create an argument called '`--no-force-download`'  to not override the Content-Type and handle it accordingly?

bkolden commented:
I've added the option as suggested, resulted in a lot of code updates.  I updated the Doc Strings for the functions updated.  Let me know what you think.  Converted positional arguments to keyword args when the count of arguments exceeded 5, which I believe is the max recommended number of positional arguments.

npes87184 commented:
> I've added the option as suggested, resulted in a lot of code updates. I updated the Doc Strings for the functions updated. Let me know what you think. Converted positional arguments to keyword args when the count of arguments exceeded 5, which I believe is the max recommended number of positional arguments.

Thank you, I will find time to review and test it. 👍

puneethkanna commented on qr_filetransfer/qr_filetransfer.py:
```diff
@@ -346,27 +374,31 @@ def print_qr_code(address):
+        file_path (str): The file path to serve.
+        **kwargs: Keyword Arguements.
 
+    Keyword Arguments:
+        debug (bool): Indication whether to output the encoded URL to the terminal.
+        custom_port (str): String indicating which custom port the user wants to use.
+        ip_addr (str): The IP address to bind web server to.
+        auth (str): Base64 encoded 'username:password'.
+        no_force_download (bool): Allow web browser to handle the file served
+            instead of forcing the browser to download it.
+    """
```
The explanation is nice, but using the brackets specifying the type might not look as good as the previous one.

debug : (Bool) Indication whether to output the encoded URL to the terminal.

I guess this would be nice. 

Just a suggestion :)

bkolden replied:
I'm using Google Style docstrings (https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html) which can optionally include the type-hints.  Personally, I like the type-hints as a reference when modifying code.  Some code linters make use of these as well.  Honestly, I'd like to add these same types of docstrings to all of the functions, but that's out of scope for this fix.  I can remove the type-hints if you like.  Please advise.

bkolden replied:
Looking at the existing docstring, I see that the data types are indeed listed, but I don't think that format would be picked up by [Sphinx](https://www.sphinx-doc.org) if we were ever to use it for generating documentation.  The Google-Style doc strings are supprted in Sphinx when using the [Napoleon extension](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html).  Not that Sphinx is being used by this project, but it allows for it if we were to choose that option.

puneethkanna replied:
It is really nice to follow the globalized pattern. Actually I don't know the pattern and 1st time seeing it. So I suggested. But we will follow the globalized pattern.

npes87184 replied:
IMO, follow this pattern is a good idea.
What do you think @sdushantha?

npes87184 commented on qr_filetransfer/qr_filetransfer.py:
```diff
@@ -397,7 +429,12 @@ def start_download_server(file_path, debug, custom_port, ip_addr, auth):
-    handler = FileTransferServerHandlerClass(file_path, auth, debug)
+    handler = FileTransferServerHandlerClass(
+        file_path,
+        kwargs.get("auth"),
```
Instead of getting it again, we can use local variable `auth`.

bkolden replied:
Updated as requested.  Good catch by the way!

npes87184 commented:
So far so good. I have some comments. The unify comment style is good if you have time, I think we can change all the comment based on this style. Also, it seen to me that the first two commits can be squashed. I expect that we will have only two commits, one for adding `guess_type`, the other for adding `--no-force-download`.

bkolden commented:
Squashed those two commits (as requested), rebased and pushed.  I've never squashed commits before, so forced me to learn something new.
I'd like to add doc strings to the remainder of the code in a separate PR, in order to separate it from this issue.

npes87184 commented:
> Squashed those two commits (as requested), rebased and pushed.  I've never squashed commits before, so forced me to learn something new.

Good to hear that you have learned something. 😀

> I'd like to add doc strings to the remainder of the code in a separate PR, in order to separate it from this issue.

OK

Thank you for your work!

From my point of view, the last three commits in this PR can also be squashed. 😉

npes87184 commented:
Hi, @bkolden 

I have changed some commit message on your second commit and added sign on commits.
You can see the modify of commit message on branch [44](https://github.com/sdushantha/qr-filetransfer/commits/44).

If you are ok on this modify. I will merge it.
Also @sdushantha would you have any suggestions on this PR?

Thanks all.

sdushantha commented:
@npes87184 This looks really good and can be merged

bkolden commented:
@npes87184 Commit message change looks good to me.

npes87184 commented:
Thanks all, this PR has been merged into `master`  branch!
